main.tf

```diff
@@ -120,11 +100,11 @@ resource "ibm_is_virtual_endpoint_gateway" "vpe" {
 resource "ibm_is_virtual_endpoint_gateway_ip" "endpoint_gateway_ip" {
   for_each = {
     # Create a map based on endpoint IP
-    for gateway_ip in local.endpoint_ip_list :
+    for gateway_ip in module.create_reserved_ips.endpoint_ip_list :
     (gateway_ip.ip_name) => gateway_ip
   }
   gateway     = local.vpe_map[each.value.gateway_name].id
-  reserved_ip = ibm_is_subnet_reserved_ip.ip[each.key].reserved_ip
+  reserved_ip = var.reserved_ips == null ? module.create_reserved_ips.reserved_ips[each.key].reserved_ip : var.reserved_ips[each.value.ip_name].reserved_ip
```

Comment:
Confused about this logic between the for_each of the block and this conditional. For example, if a reserved IP was specified by input variable, why would it also exist in the for_each of this resource block `for gateway_ip in module.create_reserved_ips.endpoint_ip_list`? I'll probably need a description of the overall logic first.